sofascoreAcademy/sofascoreAcademy/EventDetailsViewController.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  EventDetailsViewController.swift
+//  sofascoreAcademy
+//
+//  Created by Tina Jureško on 03.04.2025..
+//
+
+import Foundation
+import UIKit
+import SofaAcademic
+import SnapKit
+
+class EventDetailsViewController: UIViewController, BaseViewProtocol {
+    private let selectedEvent: EventDetailsViewModel
+    private let headerView = EventDetailsHeaderView()
+    private let customTitleView = CustomNavigationTitleView()
+    
+    init(selectedEvent: EventDetailsViewModel){
+        self.selectedEvent = selectedEvent
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        addViews()
+        setupConstraints()
+        styleViews()
+        setupNavigationBar()
+        headerView.configure(with: selectedEvent)
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        navigationController?.setNavigationBarHidden(true, animated: false)
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        navigationController?.setNavigationBarHidden(false, animated: false)
+    }
+    
+    func setupNavigationBar() {
+        guard let league = selectedEvent.league else {
+            return
+        }
+        customTitleView.configure(with: league, selectedSport: MenuViewModel.shared.selectedSport)
```

Comment:
Ovo nije dobro, što manje trebali koristiti shared varijable, osim ako baš imamo neki univerzalni service. U ovom slučaju model event koji si proslijedila ima na sebi o kojem se sportu radi. 
Event sa sebi ima teamove, ligu, sport....